### PR TITLE
Functional tests - Product settings label oos allowed/denied backorders

### PR DESCRIPTION
**Pull request by nesrineabdmouleh**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.7.x
| Description?  | Test label oos allowed/denied backorders
| Type?         | refacto
| Category?     | TE
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | no
| How to test?  | TEST_PATH="functional/BO/13_shopParameters/03_productSettings/03_productsStock/04_labelOutOfStock.js" URL_FO=yourShopURL npm run specific-test

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/18477)

### CODE REVIEW

File: tests/puppeteer/campaigns/functional/BO/13_shopParameters/03_productSettings/03_productsStock/04_labelOutOfStock.js [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+        `setLabelOutOfStock${index}`,
+        baseContext,
+      );
+      if (test.args.enable) {
+        const result = await this.pageObjects.productSettingsPage.setLabelOosAllowedBackorders(test.args.label);
+        await expect(result).to.contains(this.pageObjects.productSettingsPage.successfulUpdateMessage);
+      } else {
+        const result = await this.pageObjects.productSettingsPage.setLabelOosDeniedBackorders(test.args.label);
+        await expect(result).to.contains(this.pageObjects.productSettingsPage.successfulUpdateMessage);
+      }
```
Comment (Progi1984) [inline tests/puppeteer/campaigns/functional/BO/13_shopParameters/03_productSettings/03_productsStock/04_labelOutOfStock.js]:
```suggestion
      if (test.args.enable) {
        const result = await this.pageObjects.productSettingsPage.setLabelOosAllowedBackorders(test.args.label);
      } else {
        const result = await this.pageObjects.productSettingsPage.setLabelOosDeniedBackorders(test.args.label);
      }
      await expect(result).to.contains(this.pageObjects.productSettingsPage.successfulUpdateMessage);
```

File: tests/puppeteer/campaigns/functional/BO/13_shopParameters/03_productSettings/03_productsStock/04_labelOutOfStock.js [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+      await this.pageObjects.searchResultsPage.goToProductPage(1);
+      const lastQuantityIsVisible = await this.pageObjects.productPage.isAddToCartButtonEnabled();
+      await expect(lastQuantityIsVisible).to.be.equal(test.args.enable);
+      const availabilityLabel = await this.pageObjects.productPage.getProductAvailabilityLabel();
+      await expect(availabilityLabel).to.contains(test.args.labelToCheck);
+    });
+
+    it('should go back to BO', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', `goBackToBo${index}`, baseContext);
+      page = await this.pageObjects.productPage.closePage(browser, 1);
```
Comment (Progi1984) [inline tests/puppeteer/campaigns/functional/BO/13_shopParameters/03_productSettings/03_productsStock/04_labelOutOfStock.js:160]:
```suggestion
      await this.pageObjects.productPage.closePage(browser, 1);
```

File: tests/puppeteer/campaigns/functional/BO/13_shopParameters/03_productSettings/03_productsStock/04_labelOutOfStock.js (hunk 1 above)
Comment (nesrineabdmouleh) [reply tests/puppeteer/campaigns/functional/BO/13_shopParameters/03_productSettings/03_productsStock/04_labelOutOfStock.js]: Done :)

**Review by Progi1984 — DISMISSED** [review]

**Review by matthieu-rolland — DISMISSED** [review]

**Review by PierreRambaud — DISMISSED** [review]

**Review by Progi1984 — DISMISSED** [review]

**Review by PierreRambaud — DISMISSED** [review]

**Review by Progi1984 — DISMISSED** [review]

**Comment by Progi1984** [comment]
Hello @nesrineabdmouleh , thanks for your contribution, would you mind to rebase your pull request? Thanks

**Review by Progi1984 — APPROVED** [review]

**Review by SimonGrn — APPROVED** [review]

**Review by boubkerbribri — APPROVED** [review]

**Pull request merged**